[Ambrose Boo] iP

ambroseboo opened the pull request:
# Duke, your todo-list butler
Duke is your personal butler, keeping track of all your personal todos, deadlines, and events. 
**Features of Duke:**
1. Mark tasks as Deadlines, Todos or Events.
2. Deadlines and Events have times attached to them.
  * Deadlines are by a certain time.
  * Events are at a certain time.
3. Tasks are saved on your local disk so that you can pick up where you left off!
4. Mark Tasks as done when completed.
5. Search for Tasks by keywords.

### This project even uses javascript Enums! Just look at this code: 💯 👍 
```javascript
    public enum Type {
        TODO,
        DEADLINE,
        EVENT
    }
```
### Here are a list of implemented features and upcoming features:
- [x] Add Tasks by types of Todo, Event, Deadline
- [x] Save Tasks to local disk
- [x] Delete Task
- [x] Mark Task as done
- [x] Search for Task by keyword
- [ ] Graphical Interface 

As Lebron once said:
> This summer, I'm taking my talents to South Beach to join the Miami Heat. 
>[ew!](https://www.youtube.com/watch?v=5y_cl3WIO6A)

Take your talents to Duke this ~~summer~~ semester 1 now!!!

If you do have any improvements, please feel free leave comments in my code like so:
I think you should use an
`<ul>` element here instead.

Commit:
Add Gradle support

ambroseboo committed:
Level-1

ambroseboo committed:
Level-2

ambroseboo committed:
Level-3

ambroseboo committed:
Level-4

ambroseboo committed:
A-TextUiTesting

ambroseboo committed:
Level-5

ambroseboo committed:
Level-6

ambroseboo committed:
A-Enums

ambroseboo committed:
Add Level-7: Save

ambroseboo committed:
Add Level-8: Dates and Times

ambroseboo committed:
Add Level-8: Dates and Times

ambroseboo committed:
Add more OOP

ambroseboo committed:
Divide classes into packages

ambroseboo committed:
Add JUnit Tests

ambroseboo committed:
Add Javadocs

ambroseboo committed:
Edited to fit coding standards

ambroseboo committed:
Add find by keyword

ambroseboo committed:
Merge branch 'branch-Level-9'

ambroseboo committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

ambroseboo committed:
Merge branch 'add-gradle-support' of https://github.com/ambroseboo/ip into branch-A-Gradle

ambroseboo committed:
Delete MANIFEST.MF file

ambroseboo committed:
Fixed checkstyle errors
Ensure build and run works

ambroseboo committed:
Add full GUI functionality
Remove all text-based UI

nbvannhi commented on src/main/java/parser/Parser.java:
```diff
@@ -38,65 +38,59 @@ public Parser(Ui ui, TaskList tasks, Storage storage) {
      * @param userInput String input from user.
```
Shouldn't the first sentence in the method description start with the present tense, i.e., ```Makes``` in this case? It is also recommended to have an empty line between the description and parameter section.

nbvannhi commented on src/main/java/ui/Ui.java:
```diff
@@ -5,71 +5,60 @@
 
 public class Ui {
```
Don't forget to add a description for ```public class Ui``` too!

nbvannhi commented on src/main/java/ui/Ui.java:
```diff
@@ -5,71 +5,60 @@
 public class Ui {
     /**
-     * Prints out a welcome message to the terminal.
+     * Returns a welcome message to the terminal.
```
I like how you pay attention to details like this :)

nbvannhi reviewed (commented):
Looks good! Just some minor modifications for the comments needed.

ambroseboo committed:
Add assert tests

Add assert tests to ensure proper control flow
when parsing user input
Add assert tests to ensure proper control flow when
loading in data from file

RonaldTanSingWei reviewed (commented):
Code looks clean and good. Minor (should this commit be named level 10?) Otherwise, code looks very cohesive.

ambroseboo committed:
Improve Code Quality

Reduce indentations in load() method
Add methods in Storage to make code in load() more verbose
Add methods in Parser to make code in parseAddInput() more verbose
Add methods in Storage to make code
in markAsDoneData() more verbose

ambroseboo committed:
Merge pull request #1 from ambroseboo/branch-A-Assertions

Add assert tests

ambroseboo committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/parser/Parser.java
#	src/main/java/storage/Storage.java

ambroseboo committed:
Merge pull request #2 from ambroseboo/branch-A-CodeQuality

Improve Code Quality

ambroseboo committed:
Set up Continuous Integration

ambroseboo committed:
Set up CI

ambroseboo committed:
Code Quality improvements

Add methods in Parser class to make parse() method more concise

ambroseboo committed:
Add snooze function

Change modifiers in Parser class to make most methods private
Add function to snooze/reschedule existing tasks
Edit names of arguments in Storage class to be more verbose

ambroseboo committed:
Add JavaDocs

Add full JavaDocs to new methods in Storage class

WangGLJoseph commented:
Great work so far! The code is very saturated with documentation, making the reading experience quite pleasant. 😺 
I found it interesting that you've implemented the parse component to also handle inputs from users, and chose to interpret the datafile within the storage component instead. Might I recommend extracting out the parsing of data from storage, if you have the time? All the best! 👍

WangGLJoseph commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+     * @param input Input from user.
+     * @return Duke's generated response to user input.
+     */
+    public String getResponse(String input) {
```
Great way of abstracting out the logic of parsing the user's inputs! I think I should actually be using this too 😂 👍

WangGLJoseph commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,253 @@
+     * @return ArrayList of Tasks listed in data file.
+     * @throws IOException If reading file or file directory has issues.
+     */
+    public ArrayList<Task> load() throws IOException {
```
Might I suggest extracting out the logic for parsing from the data file to follow the Single Level of Abstraction Principle (SLAP)?

ambroseboo committed:
Edit GUI

Edit dialog box to have scrollbar to accomodate longer texts
Edit padding on dialog box
Make GUI longer horizontally

ambroseboo committed:
Add product showcase

Add product image
Add title to GUI window

ambroseboo committed:
Change Ui product photo

ambroseboo committed:
Add README

ambroseboo committed:
Set theme jekyll-theme-slate

ambroseboo committed:
Minor refactoring

Rename Ui.PNG to Ui.png
Factor out Duke.Type to enum class Type

ambroseboo committed:
Fix warnings

Ignore warning of JavaFX versioning

ambroseboo committed:
Merge remote-tracking branch 'origin/master'

ambroseboo committed:
Bug fixes to ensure compatibility with other OS

Change font to serif
Change bot image
Fix bug where data file appears in wrong folder
Minor changes to GUI

ambroseboo committed:
Final product release

Change font
Change Ui picture to reflect new UI
Minor naming adjustments
